presto-main/src/main/java/com/facebook/presto/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -310,6 +322,30 @@ public void fail()
         }
     }
 
+    @Override
+    public void setNoMorePagesForLifespan(Lifespan lifespan)
+    {
+        noMorePagesForLifespan.add(lifespan);
+    }
+
+    @Override
+    public void registerLifespanCompletionCallback(Consumer<Lifespan> callback)
```

Comment:
nit: I would reorder these methods. `registerLifespanCompletionCallback` -> `setNoMorePagesForLifespan` -> `isFinishedForLifespan`. (here in all other classes)

 `setNoMorePagesForLifespan` -> `isFinishedForLifespan` happens one after the other, so it is easier to read if they are together. And `registerLifespanCompletionCallback` happens before this two, and even before `enqueue`. So maybe it even makes sense to move `registerLifespanCompletionCallback` before the `enqueue`.

Reply:
Although I have to point out `isFull` happened before `enqueue()` :P